### PR TITLE
fix(form): outbound Mailchimp signup requests

### LINKED ISSUE #32: Form Submissions Not Reaching Mailchimp
I tried subscribing to the form on the live website, and unless there's a long delay on Mailchimp analytics, it wasn't getting the new contact.

I did use a different email to the one I subscribed with originally.

image

### PR DESCRIPTION
**Pull request by etcadinfinitum**
This fix borrows some implementation mechanics from a react module;
essentially, the URL endpoint is concatenated by hand and uses jsonp to
submit the form (so that CORS is bypassed).

Fixes #32

### CODE REVIEW

File: src/pages/home/Form.jsx [hunk 1]
```diff
@@ -32,13 +40,31 @@ const Signup = () => {
...
+							userMessage = submittedMessages.error;
+						} else if (response.result !== 'success') {
+							userMessage = submittedMessages.error;
+							console.log(response);
+						} else {
+							userMessage = submittedMessages.success;
+						}
 						hasSubmittedHandler(true);
+						// struggling to populate the response message in the form's div
+						console.log('user message is: ' + userMessage);
```
Comment (etcadinfinitum) [inline src/pages/home/Form.jsx]: I am unsure how to get the message value to populate without using an anti-pattern. @bpas247 any ideas?

File: src/pages/home/Form.jsx [hunk 2]
```diff
@@ -19,6 +20,13 @@ const SignUpSchema = Yup.object().shape({
 
 const Signup = () => {
 	const [hasSubmitted, hasSubmittedHandler] = useState(false);
+	const submittedMessages = {
```
Comment (bpas247) [inline src/pages/home/Form.jsx]: This should be declared outside of render.

File: src/pages/home/Form.jsx [hunk 3]
```diff
@@ -32,13 +40,31 @@ const Signup = () => {
...
 					/* Post to the survey form */
-					fetch('https://uwbacm.us16.list-manage.com/subscribe/post?u=8773dffc3f528b0b18fd7aef8&id=11b31196c4', {
-						method: 'post',
-						body: values,
-						mode: 'no-cors'
-					}).then((response) => {
+					jsonp(endpoint, {param: 'c'}, (err, response) => {
+						if (response.msg.includes('already subscribed')) {
+							/* do something better here */
+							userMessage = submittedMessages.duplicate;
```
Comment (bpas247) [inline src/pages/home/Form.jsx]: Mutating this variable isn’t going to cause a re-render, so if it’s being accessed in render then it should be in state.

File: src/pages/home/Form.jsx (hunk 1 above)
Comment (bpas247) [reply src/pages/home/Form.jsx]: See the above comment, which I think is related to your issue.

File: src/pages/home/Form.jsx (hunk 2 above)
Comment (etcadinfinitum) [reply src/pages/home/Form.jsx]: Ok, I moved it outside `Signup()` (I think that's what this means?)

File: src/pages/home/Form.jsx (hunk 3 above)
Comment (etcadinfinitum) [reply src/pages/home/Form.jsx]: OK, the `useState()` declarations make sense now. Thanks!!! :grin:

**Pull request merged**